cli/ts/commands/joinPoll.ts

```diff
@@ -0,0 +1,386 @@
+
+    if (transactionHash) {
+      const tx = await signer.provider!.getTransaction(transactionHash);
+      fromBlock = tx?.blockNumber ?? defaultStartBlock;
+    }
+
+    logYellow(quiet, info(`starting to fetch logs from block ${fromBlock}`));
+
+    signUpData = await genSignUpTree({
+      provider: signer.provider!,
+      address: await maciContract.getAddress(),
+      blocksPerRequest: blocksPerBatch || 50,
+      fromBlock,
+      endBlock,
+      sleepAmount: 0,
+    });
+
+    currentStateRootIndex = Number(numSignups) - 1;
+
+    if (!stateIndex) {
+      const index = signUpData.stateLeaves.findIndex((leaf) => leaf.pubKey.equals(userMaciPubKey));
+      if (index) {
+        loadedStateIndex = BigInt(index);
+      } else {
+        logError("State leaf not found");
+        process.exit();
+      }
+    }
+
+    // check < 1 cause index zero is a blank state leaf
+    if (loadedStateIndex! < 1) {
+      logError("Invalid state index");
+    }
+
+    if (!newVoiceCreditBalance) {
+      loadedCreditBalance = signUpData.stateLeaves[Number(loadedStateIndex!)].voiceCreditBalance!;
+    }
+
+    circuitInputs = joiningCircuitInputs(
+      signUpData,
+      stateTreeDepth,
+      userMaciPrivKey,
+      loadedStateIndex!,
+      loadedCreditBalance!,
+      pollPrivKeyDeserialized,
+      pollPubKey,
+    ) as unknown as CircuitInputs;
+  }
+
+  const pollVk = await extractVk(pollJoiningZkey);
+
+  try {
+    // generate the proof for this batch
+    // eslint-disable-next-line no-await-in-loop
+    const r = await genProof({
+      inputs: circuitInputs,
+      zkeyPath: pollJoiningZkey,
+      useWasm,
+      rapidsnarkExePath: rapidsnark,
+      witnessExePath: pollWitgen,
+      wasmPath: pollWasm,
+    });
+
+    // verify it
+    // eslint-disable-next-line no-await-in-loop
+    const isValid = await verifyProof(r.publicSignals, r.proof, pollVk);
+    if (!isValid) {
+      throw new Error("Generated an invalid proof");
+    }
+
+    const proof = formatProofForVerifierContract(r.proof);
+
+    // submit the message onchain as well as the encryption public key
+    const tx = await pollContract.joinPoll(
+      nullifier,
+      pollPubKey.asContractParam(),
+      loadedCreditBalance!,
+      currentStateRootIndex,
+      proof,
+    );
+    const receipt = await tx.wait();
+
+    if (receipt?.status !== 1) {
+      logError("Transaction failed");
+    }
+
+    logYellow(quiet, info(`Transaction hash: ${receipt!.hash}`));
+
+    return 0;
+  } catch (error) {
+    logError((error as Error).message);
+  }
+
+  return -1;
+};
+
+/**
+ * Parse the poll joining events from the Poll contract
+ */
+const parsePollJoinEvents = async ({
+  pollContract,
+  startBlock,
+  currentBlock,
+  pollPublicKey,
+}: IParsePollJoinEventsArgs) => {
+  // 1000 blocks at a time
+  for (let block = startBlock; block <= currentBlock; block += 1000) {
+    const toBlock = Math.min(block + 999, currentBlock);
```

Comment:
magic numbers

Reply:
it's copied from `signup.ts` - https://github.com/0x3327/maci/blob/feat/anon-poll-joining/cli/ts/commands/signup.ts#L91

I also found the same in the main repo: https://github.com/privacy-scaling-explorations/maci/blob/dev/packages/cli/ts/commands/signup.ts#L112

Reply:
So, for this file `1000` should be moved to a constant `BLOCKS_STEP`